reframe/frontend/cli.py

```diff
@@ -1393,54 +1421,65 @@ def module_unuse(*paths):
+                '--maxfail should be a non-negative integer: '
                 f'{options.maxfail}'
             )
 
         if options.reruns and options.duration:
             raise errors.CommandLineError(
-                f"'--reruns' option cannot be combined with '--duration'"
+                "'--reruns' option cannot be combined with '--duration'"
             )
 
         if options.reruns < 0:
             raise errors.CommandLineError(
-                f"'--reruns' should be a non-negative integer: {options.reruns}"
+                "'--reruns' should be a non-negative integer: "
+                f"{options.reruns}"
             )
 
         runner = Runner(exec_policy, printer, options.max_retries,
                         options.maxfail, options.reruns, options.duration)
         try:
             time_start = time.time()
-            session_info['time_start'] = time.strftime(
-                '%FT%T%z', time.localtime(time_start),
-            )
             runner.runall(testcases, restored_cases)
         finally:
+            # Build final JSON report
             time_end = time.time()
-            session_info['time_end'] = time.strftime(
-                '%FT%T%z', time.localtime(time_end)
-            )
-            session_info['time_elapsed'] = time_end - time_start
+            report.update_timestamps(time_start, time_end)
+            report.update_run_stats(runner.stats)
+            if options.restore_session is not None:
+                report.update_restored_cases(restored_cases, restored_session)
 
             # Print a retry report if we did any retries
             if options.max_retries and runner.stats.failed(run=0):
-                printer.info(runner.stats.retry_report())
+                printer.retry_report(report)
 
             # Print a failure report if we had failures in the last run
             success = True
             if runner.stats.failed():
                 success = False
-                runner.stats.print_failure_report(
-                    printer, not options.distribute,
-                    options.duration or options.reruns
+                printer.failure_report(
+                    report,
+                    rerun_info=not options.distribute,
+                    global_stats=options.duration or options.reruns
                 )
                 if options.failure_stats:
-                    runner.stats.print_failure_stats(
-                        printer, options.duration or options.reruns
+                    printer.failure_stats(
+                        report, global_stats=options.duration or options.reruns
                     )
 
             if options.performance_report and not options.dry_run:
-                printer.info(runner.stats.performance_report())
+                try:
+                    data, header = reporting.performance_compare(
+                        rt.get_option('general/0/perf_report_spec'), report
```

Comment:
For some reason this doesn't take into account the cli option value.

Reply:
This is a bug and I'm fixing it; I've noticed the same behaviour also with the existing `--timestamp` option.

Reply:
Fixed in e9f34ef6b73ade129c29ccf84a46eb3d5bec76bd.